lib/zstd.h

```diff
@@ -941,6 +710,207 @@ ZSTDLIB_API size_t ZSTD_DCtx_refPrefix(ZSTD_DCtx* dctx,
 ZSTDLIB_API size_t ZSTD_DCtx_reset(ZSTD_DCtx* dctx, ZSTD_ResetDirective reset);
 
 
+/****************************
+*  Streaming
```

Comment:
Comment, for discussion : 

"Streaming" seems pretty far in the priority order of API presentation.

There is a logic here : all one-shot compression functions are presented first, and one-shot looks like an "easier" use case than streaming. Also, presenting the advanced parameters makes them re-usable in the "streaming" section.

I'm just puzzled by the feeling that "streaming" scenarios are probably more common that "dictionary" ones. Yet, a reader will have to go pretty far down the list to discover streaming capabilities.

Should we swap this order, and have streaming before dictionary ?

It's probably not a big deal, just curious what's your take on this.